serverless/integrations/gpu-resources.adoc

```diff
@@ -0,0 +1,28 @@
+include::modules/serverless-document-attributes.adoc[]
+[id="gpu-resources"]
+= Using NVIDIA GPU resources with serverless applications
+:context: gpu-resources
+include::modules/common-attributes.adoc[]
+
+toc::[]
+
+NVIDIA support experimental use of GPU resources on {product-title}.
+See link:https://docs.nvidia.com/datacenter/kubernetes/openshift-on-gpu-install-guide/index.html[{product-title} on NVIDIA GPU accelerated clusters] for more information about setting up GPU resources on {product-title}.
+
+Once GPU resources are enabled for your {product-title} cluster, you can specify GPU requirements for a Knative service using `kn` commands.
+
+.Specifying GPU requirements for a Knative service using `kn`
+
+You can specify a GPU resource requirement when you create a Knative service using `kn`.
+
+Create a service. Set the GPU resource requirement `--limits` to 1 by using `nvidia.com/gpu=1`:
+----
+$ kn service create hello --image docker.io/mvinkler/hellocuda-go --limits nvidia.com/gpu=1
+----
+
+.Updating GPU requirements for a Knative service using `kn`
+
+Update the service. Change the GPU resource requirement `--limits` to 3 by using `nvidia.com/gpu=3`:
```

Comment:
@navidshaikh I wonder whether we should explain what changing the `--limits` but also the `--requests` of `nvidia.com/gpu` to the various values.

E.g. as a reader I would have the following questions:

* What does a limit of 1 actually mean ? That my process gets a dedicated GPU to work on ?
* Any value n larger than 1, does this mean a single pod gets started only on a node that has n GPUs installed or does this mean the my Knative service is called to n pods on separate nodes, each with 1 GPU ?
* Does the usual _soft_ vs _hard_ limit semantics apply for the GPU `--requests` and `--limits` ?
* Can only be integer values be used ? What happens when I provide a fractional number like 1.5 ? Error or Ignoring ?

I don't ever the answers as I'm not really know how the GPU support works, but I think it would be cool if we could find out  and help our users here who might have the same questions.

Reply:
@mvinkler Do you have some ideas on how we could answer these questions?

Reply:
> What does a limit of 1 actually mean ? That my process gets a dedicated GPU to work on ?

Yes, the service gets the dedicated 1 GPU resources. Service's dont share the GPU resources real time. Rest of the services needing GPU resources need to wait until the GPU resource is freed.

Currently only integer values for GPU resources are supported, no fractional values.

A limit of 1 GPU mean, restrict applications exceeding the usage more than 1 GPU resource. Your application will get a dedicated 1 GPU resource.

If requested for more than 1 GPU resource, your service is deployed on respective node where the GPU resource requirements are met. (The service pod(s) started on the node where GPU resources are equal to or more than the quantity specified in spec/CLI).

Reply:
I agree with Navid's answer. Also I found docs about quotas which apply to our `--requests` and `--limits` as well, maybe we could link it: https://docs.openshift.com/container-platform/4.4/applications/quotas/quotas-setting-per-project.html#setting-resource-quota-for-extended-resources_quotas-setting-per-project

Reply:
I am not sure if the *hard/soft* limit sematincs applies at all for GPU resource, I didnt find any docs for mentioned semantics even for cpu/memory.